Sources/Hyperspace/Service/Backend/BackendServiceProtocol.swift

```diff
@@ -28,7 +28,7 @@ public protocol BackendServiceErrorInitializable: Error {
 /// Represents the completion of a request executed using a BackendService.
 /// When successful, the parsed object is provided as the associated value.
 /// When request execution fails, the relevant E is provided as the associated value.
-public typealias BackendServiceCompletion<T, E: Swift.Error> = (Result<T, E>) -> Void
+public typealias BackendServiceCompletion<T, E: Swift.Error> = (Result<T, E>, HTTP.Response?) -> Void
```

Comment:
This might be a controversial opinion, but I don't think using the `HTTP.Response` is an important enough use case that it should get returned along with the result (and have to _ ignore it frequently). I'd much rather see us implement a way where the `Response` can be used in the `transformData` closure to create the `Result<T>`.

Reply:
I agree that the response is going to be ignored in most cases. There are cases where I need the response along with the result so I can add the result data and a value from the response header to my custom class. I am open to finding the best way to do this. Example:
`backendService.execute(request: request) { (result, response) in
            switch result {
            case .success(let successResponse):
                let items = successResponse
                let total = response?.headers["total"]
                let customClass = CustomClass(items: items, total: total)
                completion(customClass, nil)
            case .failure(let error):
                completion(nil, error)
            }
`

Reply:
What if instead we passed along the `NetworkServiceSuccess` object to the `transformData` ? This would allow you to encapsulate the logic of building your result as before, and would only require changes to the end user if you have written custom transform functions.

Reply:
I can give it a go

Reply:
@wmcginty what would `transformData` look like if passing along `NetworkServiceSuccess`?
would the service success be passed along with `data`?

Reply:
My initial thought was that `BackendServiceHelper` would change to something like: 

```swift
public static func handleResponseData<T: Request>(_ data: Data, networkSuccess: NetworkServiceSuccess, for request: T, completion: @escaping BackendServiceCompletion<T.ResponseType, T.ErrorType>) {
        let transformResult = request.transformData(data, networkSuccess) //this would need to be modified at the `Request` level as well.
        
        executeOnMain {
            switch transformResult {
            case .success(let transformedObject):
                completion(.success(transformedObject))
            case .failure(let error):
                completion(.failure(error))
            }
        }
    }

Reply:
Side note: The only reason I would suggest sending the whole success is in case something else gets added to that object in the future (as of now its just Data and HTTP.Response)

Reply:
I think I got it changed